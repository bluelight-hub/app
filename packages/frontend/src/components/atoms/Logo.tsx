import React from 'react';
import mobileLogo from "../../assets/brandbook/mobile-logo.png";
import mobileLogoWhite from "../../assets/brandbook/mobile-white.png";
import { useTheme } from "../../hooks/useTheme";
import { BaseAtomProps } from '../../utils/types';

type LogoProps = BaseAtomProps;

/**
 * Logo component that handles dark/light mode
 */
const Logo: React.FC<LogoProps> = ({
    className,
    'data-testid': dataTestId = 'logo'
}) => {
    const { isDark } = useTheme();

    return (
        <img
            src={isDark ? mobileLogoWhite : mobileLogo}
            alt="BlueLight Hub"
            className={className}
            data-testid={dataTestId}
        />
    );
};

export default Logo; 