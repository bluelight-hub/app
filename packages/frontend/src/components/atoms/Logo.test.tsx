import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import Logo from './Logo';

// Mock den useTheme Hook
vi.mock('../../hooks/useTheme', () => ({
  useTheme: () => ({
    theme: 'light',
    isDark: false,
    isAuto: false,
    setIsDark: vi.fn(),
    setAutoTheme: vi.fn(),
    toggleTheme: vi.fn(),
  }),
}));

// Mock die Bilder
vi.mock('../../assets/brandbook/mobile-logo.png', () => ({
  default: 'mocked-light-logo-path',
}));

vi.mock('../../assets/brandbook/mobile-white.png', () => ({
  default: 'mocked-dark-logo-path',
}));

describe('Logo', () => {
  // Unit Test - Grundlegendes Rendering
  it('should render without crashing', () => {
    const { container } = render(<Logo data-testid="logo" />);
    const logoElement = container.firstChild;
    expect(logoElement).toBeInTheDocument();
    expect(logoElement).toHaveAttribute('alt', 'BlueLight Hub');
  });

  // Unit Test - Custom className wird korrekt angewendet
  it('should apply custom className', () => {
    render(<Logo className="custom-logo-class" data-testid="logo" />);
    const logoElement = screen.getByTestId('logo');
    expect(logoElement).toHaveClass('custom-logo-class');
  });

  // Integration Test - Verhalten im Layout
  it('should work properly in a layout context', () => {
    render(
      <div data-testid="container">
        <header>
          <Logo data-testid="logo" />
        </header>
      </div>,
    );

    const container = screen.getByTestId('container');
    const logoElement = screen.getByTestId('logo');
    expect(container).toContainElement(logoElement);
  });

  // Snapshot Test
  it('should match snapshot', () => {
    const { container } = render(<Logo data-testid="logo" />);
    expect(container).toMatchSnapshot();
  });
});
