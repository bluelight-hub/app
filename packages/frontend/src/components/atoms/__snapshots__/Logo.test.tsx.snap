// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo > should match snapshot 1`] = `
<div>
  <img
    alt="BlueLight Hub"
    data-testid="logo"
    src="mocked-light-logo-path"
  />
</div>
`;
